explorer/client/src/api/v1/admin/auth.ts

```diff
@@ -1,32 +1,39 @@
 import * as jsonapi from '@chainlink/json-api-client'
-
+import { Api } from '@chainlink/json-api-client'
+import { boundMethod } from 'autobind-decorator'
 const SIGN_IN_ENDPOINT = '/api/v1/admin/login'
 
 interface SignInRequestParams {
   username: string
   password: string
 }
 
-/**
- * SignIn authenticates an admin user.
- * @param username The admin username
- * @param password The plain text password
- */
-export function signIn(
-  username: string,
-  password: string,
-): Promise<jsonapi.PaginatedApiResponse<number>> {
-  return jsonapi.createResource<SignInRequestParams, number>(SIGN_IN_ENDPOINT)({
-    username,
-    password,
-  })
-}
-
 const SIGN_OUT_ENDPOINT = '/api/v1/admin/logout'
 
-/**
- * SignOut authenticates an admin user.
- */
-export function signOut(): Promise<{}> {
-  return jsonapi.deleteResource<{}, {}>(SIGN_OUT_ENDPOINT)()
+export class Auth {
+  constructor(private api: Api) {}
+
+  /**
+   * SignIn authenticates an admin user.
+   * @param username The admin username
+   * @param password The plain text password
+   */
+  @boundMethod
+  public signIn(
+    username: string,
+    password: string,
+  ): Promise<jsonapi.PaginatedApiResponse<number>> {
```

Comment:
Is signIn returns a PaginatedApiResponse type?

Reply:
It definitely does not, good catch. Mind opening a ticket for this since it's out of scope of this PR?